libyul/AsmAnalysis.cpp

```diff
@@ -308,31 +309,48 @@ vector<YulString> AsmAnalyzer::operator()(FunctionCall const& _funCall)
 	vector<YulString> const* returnTypes = nullptr;
 	vector<optional<LiteralKind>> const* literalArguments = nullptr;
 
-	if (BuiltinFunction const* f = m_dialect.builtin(_funCall.functionName.name))
+	auto inScope = [&](YulString const& _function)
 	{
-		parameterTypes = &f->parameters;
-		returnTypes = &f->returns;
-		if (!f->literalArguments.empty())
-			literalArguments = &f->literalArguments;
+		return m_currentScope->lookup(_function, GenericVisitor{
+			[&](Scope::Variable const&)
+			{
+				m_errorReporter.typeError(
+					4202_error,
+					nativeLocationOf(_funCall.functionName),
+					"Attempt to call variable instead of function."
+				);
+			},
+			[&](Scope::Function const& _fun)
+			{
+				parameterTypes = &_fun.arguments;
+				returnTypes = &_fun.returns;
+			}
+		});
+	};
+
+	// TODO remove this in 0.9.0. We allow creating functions or identifiers in Yul with the name
```

Comment:
Do you think it would be a good idea to open an issue preemptively, so we don't forget, or is this something that will actively cause problems once we bump to 0.9.0. and will have to be removed anyway?